source/extensions/filters/http/oauth2/filter.cc

```diff
@@ -509,6 +543,57 @@ void OAuth2Filter::finishGetAccessTokenFlow() {
   config_->stats().oauth_success_.inc();
 }
 
+void OAuth2Filter::finishUpdateAccessTokenFlow() {
+  ASSERT(config_->useRefreshToken());
+  // update cookie header to send it on actual state to upstream
+  const CookieNames& cookie_names = config_->cookieNames();
+
+  absl::flat_hash_map<std::string, std::string> cookies =
```

Comment:
I don't think you need set those cookies in request to response_headers_to_add again, just new cookies should be fine.

Reply:
You suggest to not save new cookies to the response_headers_to_add?
I tried it before, but it doesn't works. I didn't deep why it is. So I just decided to save cookies and set cookies through set-cookie headers in the response

Reply:
I'm suggesting not set all `Cookie` in request header to `Set-Cookie` as this might cause a security issue that a new `Set-Cookie` can overwrite the original expiration, or remove some property like `Secure` or `HttpOnly`.

Reply:
`Set-Cookie` is applied only for specific cookies, related with OAuth2: id Token, RefreshToken, AccessToken, OAuth Hmac, OAuth Expires. Other cookies which recieved in the request I resend in initial state. I cannot add additional header `Cookie` in the request. As I know, only one header `Cookie` in the request is allowed. Looks like I don't catch your suggestion, I am sorry. 
Could you explain again please what I need to do?

Reply:
My understanding (and maybe @lizan can clarify if I'm misrepresenting what he's saying) is that you don't need to include all the cookies that are present on the request headers in the response headers, only the ones that you are trying to update. That way you are not resetting more cookies than necessary

Reply:
@snowp , where do you see I touch other cookies except only the ones that are related with oauth2? Take a look please `finishRefreshAccessTokenFlow` method.
I only update `cookies oauth_hmac, oauth_expires, bearer_token, id_token, refresh_token` cookies.